src/hasher.rs

```diff
@@ -41,7 +41,7 @@ pub trait Digest<const S: usize>:
         use unsigned_varint::io::read_u64;
 
         let size = read_u64(&mut r)?;
-        if size > S as u64 || size > u8::max_value() as u64 {
+        if size > S as u64 || size > u16::MAX as u64 {
```

Comment:
Do we really want to increase this? Isn't 256 bytes (2048 bit) enough for hashes. I know it limits the identity hash, but even for that it's probably good enough. I also haven't heard any complaints about the size being too small. I'd keep it to 256 bytes.

(It's the same for other occurences)